packages/core/src/Popup.js

```diff
@@ -110,44 +106,52 @@ function Popup({
     icon,
     message,
     buttons,
+    buttonsDirection,
     // React props
     className,
     children,
-    ...popupProps
+    ...popupProps,
 }: Props) {
     const rootClassName = classNames(BEM.root.toString(), className);
 
     return (
         <div className={rootClassName} {...popupProps}>
             <Overlay />
 
-            <div className={BEM.container.toString()}>
-                <div className={BEM.body.toString()}>
-                    {renderPopupIcon(icon)}
-                    {renderPopupMessage(message)}
+            <div className={BEM.container}>
+                <div className={BEM.body}>
+                    {icon && wrapIfNotElement(icon, { with: PopupIcon, via: 'type' })}
+                    {message && wrapIfNotElement(message, { with: PopupMessage, via: 'text' })}
                 </div>
 
-                {renderPopupButtons(buttons)}
+                {renderPopupButtons(buttons, buttonsDirection)}
                 {children}
             </div>
         </div>
     );
 }
 
+const StringOrElement = PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.element,
+]);
+
 Popup.propTypes = {
-    icon: PropTypes.node,
-    message: PropTypes.node,
-    buttons: PropTypes.arrayOf(PropTypes.element)
+    icon: StringOrElement,
+    message: StringOrElement,
+    buttons: PropTypes.arrayOf(PropTypes.element),
+    buttonsDirection: PropTypes.oneOf(Object.values(BUTTONS_DIRECTION)),
 };
 
 Popup.defaultProps = {
     icon: null,
     message: null,
-    buttons: []
+    buttons: [],
+    buttonsDirection: BUTTONS_DIRECTION.VERTICAL,
 };
 
 // export for tests
 export { Popup as PurePopup };
-export const EscapablePopup = closable({ onEscape: true })(Popup);
+export const EscapablePopup = closable({ onEscape: false })(Popup);
```

Comment:
Since the `onEscape` option was disabled, that means `< EscapablePopup>` became meaningless.

How about remove this HOC by default? We still can apply it outside of `<Popup>` component when we actual need this behavior. But the another question is mixin components are exportable or only use in gypcrete? XD

Reply:
I was thinking of the advantage of keeping `closable()` mixin, because it supports runtime config now and you can turn it on/off easily. I think you're right on this: since it's not default behavior, it can totally be removed from `<Popup>`.

I'm not sure about the second question yet. What do you think if we export all the mixins and utils to the bundle, maybe like this way:

```js
import { icBEM } from '@ichef/gypcrete/utils`
import { closable } from '@ichef/gypcrete/mixins`
```

Reply:
The runtime options are a little bit magic, they cannot be read by the propTypes of `<Popup>` component, unless we read the source codes lol. So I think remove it from `<Popup>` is better, but need to make sure the closable behavior is not usually used in the iCHEF codebase.

For second question, maybe we can move the utils, mixins or helper functions into a package? That they can be easily reuse in another places.

Reply:
I'm not sure if they're really reusable outside the gypcrete context... XD

Reply:
The exportation problem we can discuss later, it should not be a scope of this PR XDD

Lets we fix the `closable()` mixin issue in popup component first, do you think remove it is the better ideal?

Reply:
yeah let's remove it